### PR TITLE
BUG: DatetimeIndex.is_year_start breaks on custom business days frequencies bigger then `1C`

### LINKED ISSUE #58664: BUG: DatetimeIndex.is_year_start breaks on custom business days frequencies bigger then `1C`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

dr1 = pd.date_range("2020-01-01", periods=1, freq="2C")
print(dr1.is_year_start)

dr2 = pd.date_range("2020-01-01", periods=1, freq="C")
print(dr2.is_year_start)
```

### Issue Description

This outputs:
```
[ True]
........
ValueError: Custom business days is not supported by is_year_start
```

### Expected Behavior

it should raise `ValueError` in both cases
```
ValueError: Custom business days is not supported by is_year_start
```

### Installed Versions

<details>

commit                : ae1928b0aac0ebcefe2158891c8a837bab242b9b
python                : 3.10.13.final.0
python-bits           : 64
OS                    : Linux
OS-release            : 4.19.0-22-amd64
Version               : #1 SMP Debian 4.19.260-1 (2022-09-29)
machine               : x86_64
processor             : 
byteorder             : little
LC_ALL                : None
LANG                  : en_GB.UTF-8
LOCALE                : en_GB.UTF-8

pandas                : 3.0.0.dev0+478.gae1928b0aa.dirty
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0
setuptools            : 69.1.1
pip                   : 24.0
Cython                : 3.0.8
pytest                : 8.0.2
hypothesis            : 6.98.17
sphinx                : 7.2.6
blosc                 : None
feather               : None
xlsxwriter            : 3.1.9
lxml.etree            : 5.1.0
html5lib              : 1.1
pymysql               : 1.4.6
psycopg2              : 2.9.9
jinja2                : 3.1.3
IPython               : 8.22.2
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
bottleneck            : 1.3.8
fastparquet           : 2024.2.0
fsspec                : 2024.2.0
gcsfs                 : 2024.2.0
matplotlib            : 3.8.3
numba                 : 0.59.0
numexpr               : 2.9.0
odfpy                 : None
openpyxl              : 3.1.2
pyarrow               : 15.0.0
pyreadstat            : 1.2.6
python-calamine       : None
pyxlsb                : 1.0.10
s3fs                  : 2024.2.0
scipy                 : 1.12.0
sqlalchemy            : 2.0.27
tables                : 3.9.2
tabulate              : 0.9.0
xarray                : 2024.2.0
xlrd                  : 2.0.1
zstandard             : 0.22.0
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #58664
- [x] Tests added and passed
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst

### CODE REVIEW

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -419,6 +419,7 @@ Interval
 Indexing
 ^^^^^^^^
 - Bug in :meth:`DataFrame.__getitem__` returning modified columns when called with ``slice`` in Python 3.12 (:issue:`57500`)
+- Bug in :meth:`DatetimeIndex.is_year_start` and :meth:`DatetimeIndex.is_quarter_start` does not raise on Custom business days frequencies bigger then "1C" (:issue:`58664`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:422]: Could you put this under `Datetimelike`?

File: pandas/_libs/tslibs/fields.pyx [hunk 2]
```diff
@@ -243,18 +243,17 @@ def get_start_end_field(
...
+        is_business = freq_name[0] == "B"
 
         # YearBegin(), BYearBegin() use month = starting month of year.
         # QuarterBegin(), BQuarterBegin() use startingMonth = starting
         # month of year. Other offsets use month, startingMonth as ending
         # month of year.
 
-        if (freqstr[0:2] in ["MS", "QS", "YS"]) or (
-                freqstr[1:3] in ["MS", "QS", "YS"]):
+        if freq_name.lstrip("B")[0:2]  in ["MS", "QS", "YS"]:
```
Comment (mroeschke) [inline pandas/_libs/tslibs/fields.pyx:256]:
```suggestion
        if freq_name.lstrip("B")[0:2] in ["MS", "QS", "YS"]:
```

**Comment by MarcoGorelli** [comment]
this needs rebasing, as it contains commits from the other PR (and could you move the whatsnew from that one to `DatetimeLike` please?)

File: doc/source/whatsnew/v3.0.0.rst (hunk 1 above)
Comment (natmokval) [reply doc/source/whatsnew/v3.0.0.rst]: thanks, updated

File: pandas/_libs/tslibs/fields.pyx (hunk 2 above)
Comment (natmokval) [reply pandas/_libs/tslibs/fields.pyx]: done

**Comment by natmokval** [comment]
> this needs rebasing, as it contains commits from the other PR (and could you move the whatsnew from that one to `DatetimeLike` please?)
> 
thanks, rebased and moved the note in whatsnew from `Indexing` to `DatetimeLike`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @natmokval

**Pull request merged**